src/gmp_backend.rs

```diff
@@ -223,11 +237,20 @@ impl Bn {
 
     /// Convert this big number to a big-endian byte sequence
     pub fn to_bytes(&self) -> Vec<u8> {
+        debug_assert!(self >= &Self::zero());
+
+        if self < &Self::zero() {
+            return vec![0];
+        }
+
         let mut s = self.0.to_str_radix(16);
+
         if s.len() & 1 == 1 {
             s = format!("0{}", s);
         }
-        hex::decode(&s).unwrap()
+
+        // TODO: Use a bad default instead of panic. This should be replaced with a nicer function
```

Comment:
I don't understand this comment.  Aren't you already using a "bad" default (ie. 0)?  What would a "nicer function" accomplish?

Reply:
Sorry, by "this" I meant the `to_bytes` function should be replaced with one that can't panic.